go/vt/vtgate/vindexes/vschema_test.go

```diff
@@ -247,57 +247,30 @@ func TestUnshardedVSchemaValid(t *testing.T) {
 		Vindexes: make(map[string]*vschemapb.Vindex),
 		Tables:   make(map[string]*vschemapb.Table),
 	})
-	if err != nil {
-		t.Errorf("TestUnshardedVSchemaValid:\n%v", err)
-	}
+	require.NoError(t, err)
 }
 
 func TestUnshardedVSchema(t *testing.T) {
 	good := vschemapb.SrvVSchema{
 		Keyspaces: map[string]*vschemapb.Keyspace{
 			"unsharded": {
 				Tables: map[string]*vschemapb.Table{
-					"t1": {},
-				},
-			},
-		},
-	}
+					"t1": {}}}}}
+
 	got := BuildVSchema(&good)
-	err := got.Keyspaces["unsharded"].Error
+	require.NoError(t, got.Keyspaces["unsharded"].Error)
+
+	table, err := got.FindTable("unsharded", "t1")
 	require.NoError(t, err)
-	ks := &Keyspace{
-		Name: "unsharded",
-	}
-	t1 := &Table{
-		Name:     sqlparser.NewIdentifierCS("t1"),
-		Keyspace: ks,
-	}
-	dual := &Table{
-		Name:     sqlparser.NewIdentifierCS("dual"),
-		Keyspace: ks,
-		Type:     TypeReference,
-	}
-	want := &VSchema{
-		RoutingRules: map[string]*RoutingRule{},
-		uniqueTables: map[string]*Table{
-			"t1":   t1,
-			"dual": dual,
```

Comment:
it is unclear why so many assertions are removed from this test file

Reply:
we were using `!reflect.DeepEqual(got, want)` in a lot of places where I felt it was better to have test that exercise methods instead of testing the inner state of structs. testing the state makes the tests extremely fragile - a change in how we store the data, like this PR does, forces us to have to change ALL assertions. Instead, we want to keep the _behaviour_ of the code, not which internal state we use to achieve this behaviour.

My intention was not to just remove assertions, I wanted to replace them with behaviour tests.